homeassistant/components/modbus/sensor.py

```diff
@@ -125,7 +125,10 @@ async def async_update(self, now: datetime | None = None) -> None:
         if self._coordinator:
             if result:
                 result_array = list(
-                    map(float if self._precision else int, result.split(","))
+                    map(
+                        float if self._precision and not self._value_is_int else int,
+                        result.split(","),
+                    )
```

Comment:
May be you should move the the type logic to a separate function to make this less complex.

Reply:
That would be a lot of overhead, with parameter etc, BUT I have simplified it, because the test for _precision is not needed.